lib/harness/vagrant.js

```diff
@@ -0,0 +1,62 @@
+const SSHConnector = require( "./ssh" );
+const path = require( 'path' );
+const bakerForMacPath = process.platform === 'darwin' ? path.join( require( 'os' ).homedir(), 'Library', 'Baker', 'BakerForMac' ) : undefined;
+const privateKey = process.platform === 'darwin' ? path.join( bakerForMacPath, 'baker_rsa' ) : path.join( boxes, 'baker_rsa' );
+const cmd = require( 'node-cmd' );
+
+class VagrantConnector extends SSHConnector {
+
+    constructor() {
+        super( 'vagrant@', privateKey );
+    }
+
+    async getName( context ) {
+        return context.bakerPath;
+    }
+
+    async tearDown( obj ) {
+        if ( obj && obj.child ) {
+            // 'SIGINT'
+            console.log( `\tTearing down` );
+            obj.child.stdout.removeAllListeners( "data" );
+            obj.child.stdin.write( '\x03' );
+            obj.child.kill();
+        }
+    }
+
+    async sleep( ms ) {
+        return new Promise( resolve => setTimeout( resolve, ms ) );
+    }
+
+    async getSshConfig() {
+        var vagrantData = null;
+
+        cmd.get(
+            'vagrant ssh-config',
+            function ( err, data, stderr ) {
+                vagrantData = data;
+            }
+        );
+
+        await sleep( 4000 );
```

Comment:
I think it will be better to use a promise directly on the `child_process.exe()` function, to get the output of the shell command. This may not work or may be inefficient in other cases (might take less that or more than 4 seconds to return).